ProjectLibertyLabs/gateway
# Implement revokeDelegation(ByDelegator|ByProvider)

# Description
As a consumer of the Gateway API, I want to be able to revoke a previously granted delegation

Should implement:

- [x] `GET /account/accounts/revokeDelegation/{accountId}/{providerId}` 
  - [x] (returns a `revokeDelegationByDelegator` encoded extrinsic that can be signed)
- [x] `POST /account/accounts/revokeDelegation` submits a signed extrinsic for execution

Closes #411 

# Solution

New endpoints were added to implement the desired functionality.

## Steps to Verify:

1. Execute `./scripts/account/restart-chain-docker.sh` and make sure `account-service` is running. You will need to additionally run the following command to start the account service, or start it locally, as the restart-chain script no longer brings up account.
  - `docker compose -f docker-compose.yaml -f docker-compose-e2e.account.yaml --profile e2e --profile account up -d`
3. Execute `cd ./apps/account-api/test/setup/ && npm run test-revoke-e2e`
4. Verify that `msa.DelegationRevoked` event appears on the local node.
Note: This will only work once and then the chain state needs to be restarted.
